### DISCUSSION TITLE
Undocumented breaking change in hover?

### DISCUSSION THREAD

**Discussion by Parnswir** [testing-library/user-event#944] [General]
Hi there,

we've recently upgraded to v14 from v13.5 and experienced a breaking test that seems to be an undocumented breaking change. I'm starting this discussion to make sure if it was intended or not. I don't have a reproduction repo (yet), so please bear with me.

The environment is Nextjs / next-jest / jsdom. We have a Tippy (tooltip on hover) on a container element like this:

```
<Tippy content={<div>Tooltip Content</div>}>
  <div>
    <div>
      <span>Nested Content</span>
    </div>
  </div>
</Tippy>
```
=> hovering on "Nested Content" or any of its parents triggers a tooltip "Tooltip Content" to be shown.

We tested it like this in v13.5:
```
it("should show a tooltip when hovering", async () => {
  const element = screen.getByText("Nested Content");
  userEvent.hover(element);
  expect(await screen.findByText("Tooltip Content")).toBeInTheDocument();
});
``` 

In v14 it should be like this:
```
it("should show a tooltip when hovering", async () => {
  const element = screen.getByText("Nested Content");
  await userEvent.hover(element);
  expect(await screen.findByText("Tooltip Content")).toBeInTheDocument();
});
```

But this actually fails to trigger the tooltip.

If we use `.parentElement`, the test passes again:

```
it("should show a tooltip when hovering", async () => {
  const element = screen.getByText("Nested Content");
  await userEvent.hover(element.parentElement.parentElement);
  expect(await screen.findByText("Tooltip Content")).toBeInTheDocument();
});
```

I noticed that `hover` used to iterate over the target's parent elements [in v13.5](https://github.com/testing-library/user-event/blob/v13.5.0/src/hover.ts#L35) but doesn't seem to do that in v14. I don't see any mention of this in the docs, so was that intended behavior that broke with the new version or was it never intended to behave that way?

**Comment by ph-fritsche** [comment]
With the previous version the user could not express moving from one element to another.
v14 introduced the `.pointer()` API which tries to emulate pointer device behavior. The implementations of the pointer-related APIs have been replaced by internal calls to `.pointer()`.
Just copying the iteration would have been a bug in the new API.
`pointerenter` is not dispatched on all parents when moving over an element, but on each element `when a pointing device is moved into the hit test boundaries of an element or one of its descendants`.

Finding the closest common ancestor when moving from one element to another, and dispatching `enter` respectively `leave` on the elements in between might be an appropriate implementation.
But edge cases must be considered before bringing in such a feature, as we don't want to break it / require a major version bump.

An issue collecting information about this and discussing the constraints of this or any other implementation would be welcome.

https://codesandbox.io/s/pointer-enter-leave-dom-mgzuzv
